src/libdredd/src/mutation_replace_binary_operator.cc

```diff
@@ -351,16 +352,19 @@ void MutationReplaceBinaryOperator::GenerateArgumentReplacement(
   // RHS
   // These cases are equivalent to constant replacement with the respective
   // constants
-  clang::Expr::EvalResult rhs_eval_result;
-  bool rhs_is_int =
-      binary_operator_.getRHS()->EvaluateAsInt(rhs_eval_result, ast_context);
   if (!optimise_mutations ||
-      !(rhs_is_int && (llvm::APSInt::isSameValue(rhs_eval_result.Val.getInt(),
-                                                 llvm::APSInt::get(0)) ||
-                       llvm::APSInt::isSameValue(rhs_eval_result.Val.getInt(),
-                                                 llvm::APSInt::get(1)) ||
-                       llvm::APSInt::isSameValue(rhs_eval_result.Val.getInt(),
-                                                 llvm::APSInt::get(-1))))) {
+      !(MutationReplaceExpr::ExprIsEquivalentToInt(*binary_operator_.getRHS(),
+                                                   0, ast_context) ||
+        MutationReplaceExpr::ExprIsEquivalentToFloat(*binary_operator_.getRHS(),
+                                                     0, ast_context) ||
```

Comment:
Similarly here, and in the code below for the RHS, please use double constants where relevant.

Reply:
Done.